STYLEGUIDE.md

```diff
@@ -0,0 +1,25 @@
+## Conda Style Guide
+
+In an effort to decrease the context switching/learning barrier between the different projects within the Conda org, a standard set of code styles is defined. The intention of these guidelines is not to overburden or complicate the contribution process; if any style starts to become a burden, its inherent benefit should be reevaluated.
+
+> ⚠️ See conda-incubator/ceps#24 for our policy regarding the Python versions Conda projects need to support. Conda plugins must also support this range of versions in order to continue to be of value to current users.
+
+| | Policy | Tooling |
+|---|---|---|
+| <a name="0"></a>0 | Use `black` (or `darker` for preexisting projects). | [`black`](https://github.com/psf/black) [`darker`](https://github.com/akaihola/darker) |
+| <a name="1"></a>1 | Use `isort`. | [`isort`](https://github.com/PyCQA/isort) |
+| <a name="2"></a>2 | Prefer f-strings over all other string formatting styles (e.g. `str.format`, c-style). | [`flake8-use-fstring`](https://github.com/MichaelKim0407/flake8-use-fstring) |
+| <a name="3"></a>3 | Prefer `pathlib` over `os.path`. | [`flake8-use-pathlib`](https://gitlab.com/RoPP/flake8-use-pathlib) |
+| <a name="4"></a>4 | Use typing (or gradually enforce for preexisting projects). | [`flake8-annotations`](https://github.com/sco1/flake8-annotations) |
+| <a name="4.1"></a>4.1 | Define typing aliases at the top of files. | |
+| <a name="4.2"></a>4.2 | Use `TYPE_CHECKING` import guard. | [`flake8-typing-imports`](https://github.com/asottile/flake8-typing-imports)
+| <a name="5"></a>5 | Define loggers at the top of files. | |
+| <a name="6"></a>6 | Only use `assert` in tests. | [`flake8-useless-assert`](https://github.com/decorator-factory/flake8-useless-assert) |
+| <a name="7"></a>7 | Use American English. | [`flake8-spellcheck`](https://github.com/MichaelAquilina/flake8-spellcheck) |
+| <a name="7.1"></a>7.1 | Use descriptive variable names: <ul><li>no one-char variables</li><li>avoid abbreviations</li><li>avoid overloading builtins</li></ul> | [`flake8-variables-names`](https://github.com/best-doctor/flake8-variables-names) |
+| <a name="8"></a>8 | When indicating a change that requires a future version of Python (or dropping a currently-supported version), use the following format:<br>`# FUTURE: <minimum Python requirement>, <details>`<br>e.g.:<br>`# FUTURE: Python 3.9+, replace with ...` | |
```

Comment:
I'm not sure about that, one thing that Python, Django etc uses are the [`.. versionadded` and `.. versionchanged` Sphinx directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded), which Sphinx' autodoc can read in docstrings and next to module level variables. 

Could you look into if there is a pattern that is Sphinx/autodoc compatible that would allow us to bubble up the "future" information in the docs as well? Like a `.. future: Python 3.9+, replace with ...` for docstrings and `#: .. future: Python 3.9+, replace with ...` for [module or class attribute documentation comments](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoproperty)?

Reply:
I'm all for reusing existing standards, but I'm not sure we want this kind of info bubbled up into the docstrings. This seems too developer-focused, not something the end-user would care about (and could even be confused by).

My intention with the formatting I chose was to mirror that of `# TODO` because it's something to do in the future. The goal was to make a syntax that made it an easy find/replace process.

This was originally inspired by trying to indicate future tech-debt work:

```python
# FUTURE: Python 3.8+, replace w/ functools.cache_property
@property
@functools.lru_cache(maxsize=None)
def plugin_manager(self):
    ...

# FUTURE: Python 3.9+, replace w/ functools.cache
@functools.lru_cache(maxsize=None)
def _get_cpu_info():
    ...
```

Reply:
We agreed to drop this policy and brainstorm further in a new PR.